src/librustc_mir/build/mod.rs

```diff
@@ -238,7 +238,8 @@ pub fn construct_const<'a, 'gcx, 'tcx>(hir: Cx<'a, 'gcx, 'tcx>,
     let span = tcx.map.span(item_id);
     let mut builder = Builder::new(hir, span);
 
-    let extent = ROOT_CODE_EXTENT;
+    let extent = tcx.region_maps.temporary_scope(ast_expr.id)
```

Comment:
> Existing constant expression testcases started failing. The 1+2 temporary in [T; 1+2+3] has a scope that's the function's. You think I should dig deeper into middle::region and make it the same as if it was outside of a functions?

No, just trying to understand where this change came from; it seemed kind of out of the blue. I'm a bit confused still though. What is `[T; 1+2+3]`? Some expression in a functon?